tests/unit/test_scraper.py

```
@@ -689,4 +721,412 @@
+        
+def test_scraper_performance_metrics_timeout():
+    """タイムアウト時のパフォーマンスメトリクステスト"""
+    scraper = Scraper()
+    url = "https://example.com"
+    
+    def slow_response(*args, **kwargs):
+        time.sleep(2)  # 2秒待機
+        return MagicMock(text="<div>Response</div>")
+        
+    with patch('requests.get', side_effect=slow_response):
+        with pytest.raises(requests.exceptions.Timeout):
```

Ruff (0.8.2)

Undefined name `requests`

(F821)